library/common-library/templates/_serviceaccount.tpl

```diff
@@ -47,9 +47,9 @@
 {{- end -}}
 
 {{- if (include "common.serviceAccount.create" .) -}}
-    {{- default (include "common.naming.fullname" .) $globalServiceAccount $localServiceAccount -}}
+    {{- default (include "common.naming.fullname" .) $localServiceAccount $globalServiceAccount -}}
```

Comment:
helm function are always confusing since` {{param1 | func param2 }}` and `{{ func param2 param1 }}` follows different order (at least in my head) 

Isn't crearer using pipes with multiple parameters?

Reply:
I also see this as a good measure to increase readability.

This is the reason I failed and this issue you are fixing exists.

Reply:
sure I can go for it

Reply:
{{- $globalServiceAccount | default ($localServiceAccount | default (include "common.naming.fullname" .)) -}}
or
{{- $globalServiceAccount | default ("common.naming.fullname" $localServiceAccount) -}}

Reply:
`{{- $globalServiceAccount | default $localServiceAccount | default (include "common.naming.fullname" .) -}}`

Reply:
That last one works.. it's the best option.

Reply:
yes I think the same, I believe it should be `{{- $localServiceAccount | default $globalServiceAccount | default (include "common.naming.fullname" .) -}}`, unless helm tricks me again 😄

Reply:
I am putting the global as the most important one, then the local and then the default. it's intended, and it's what I did in the first change is it incorrect? does the local have to be more important than the global?

Reply:
Oks I changed the order and added extra tests

Reply:
The most important is the local. In case local does not exist then the global should default. Then default to fullname/`default`.

Reply:
Yes, it's fixed now